src/aot/CMakeLists.txt

```diff
@@ -1,9 +1,6 @@
 add_library(aot STATIC aot.cpp)
-add_dependencies(aot parser version_h)
-target_link_libraries(aot required_resources)
-target_include_directories(aot PUBLIC ${CMAKE_SOURCE_DIR}/src)
-target_include_directories(aot PUBLIC ${CMAKE_BINARY_DIR}/src)
-target_include_directories(aot PUBLIC ${CMAKE_BINARY_DIR})
+add_dependencies(aot version_h)
+target_link_libraries(aot required_resources parser)
```

Comment:
Why is `parser` moved here?

Reply:
add_dependencies was not necessary for `parser`: https://stackoverflow.com/questions/74043615/when-should-i-use-cmakes-add-dependencies-instead-of-target-link-libraries

The link dependency is enough. However for version_h, it's just a header file, so need to manually set the dep

Reply:
Makes sense, thanks.